src/shared/metadata/cgroup_path_resolver.cc

```diff
@@ -52,12 +54,18 @@ StatusOr<std::string> CGroupBasePath(std::string_view sysfs_path) {
   auto fs_status = statfs(cgv2_base_path.c_str(), &info);
   bool cgroupv2 = (fs_status == 0) && (info.f_type == CGROUP2_SUPER_MAGIC);
 
-  if (cgroupv2 || FLAGS_force_cgroup2_mode) {
-    return cgv2_base_path;
+  if (cgroupv2 || FLAGS_test_only_force_cgroup2_mode) {
+    if (FLAGS_test_only_force_cgroup2_mode) {
+      return std::vector{cgv2_base_path};
+    }
+    base_paths.push_back(cgv2_base_path);
   }
   // (TODO): This check for cgroup2FS is eventually to be moved above the cgroupv1 check.
 
-  return error::NotFound("Could not find CGroup base path");
+  if (base_paths.empty()) {
+    return error::NotFound("Could not find CGroup base path");
+  }
+  return base_paths;
```

Comment:
minor: Could consider returning the empty list and letting the caller deal with this, now that this returns a list of candidates.